src/Core/Addon/AddonManagerHelper.php

```diff
@@ -0,0 +1,306 @@
+<?php
+
+// Copyright (C) 2010-2024, the Friendica project
+// SPDX-FileCopyrightText: 2010-2024 the Friendica project
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+declare(strict_types=1);
+
+namespace Friendica\Core\Addon;
+
+use Friendica\Core\Cache\Capability\ICanCache;
+use Friendica\Core\Config\Capability\IManageConfigValues;
+use Friendica\Database\Database;
+use Friendica\Util\Profiler;
+use Friendica\Util\Strings;
+use Psr\Log\LoggerInterface;
+
+/**
+ * helper functions to handle addons
+ *
+ * @internal
+ */
+final class AddonManagerHelper implements AddonHelper
+{
+	private string $addonPath;
+
+	private Database $database;
+
+	private IManageConfigValues $config;
+
+	private ICanCache $cache;
+
+	private LoggerInterface $logger;
+
+	private Profiler $profiler;
+
+	/** @var string[] */
+	private array $addons = [];
+
+	public function __construct(
+		string $addonPath,
+		Database $database,
+		IManageConfigValues $config,
+		ICanCache $cache,
+		LoggerInterface $logger,
+		Profiler $profiler
+	) {
+		$this->addonPath = $addonPath;
+		$this->database  = $database;
+		$this->config    = $config;
+		$this->cache     = $cache;
+		$this->logger    = $logger;
+		$this->profiler  = $profiler;
+	}
+	/**
+	 * Returns the absolute path to the addon folder
+	 *
+	 * e.g. `/var/www/html/addon`
+	 */
+	public function getAddonPath(): string
+	{
+		return $this->addonPath;
+	}
+
+	/**
+	 * Returns the list of available addons.
+	 *
+	 * This list is made from scanning the addon/ folder.
+	 * Unsupported addons are excluded unless they already are enabled or system.show_unsupported_addon is set.
+	 *
+	 * @return string[]
+	 */
+	public function getAvailableAddons(): array
+	{
+		$dirs = scandir($this->getAddonPath());
+
+		if (!is_array($dirs)) {
+			return [];
+		}
+
+		$files = [];
+
+		foreach ($dirs as $dirname) {
+			// ignore hidden files and folders
+			// @TODO: Replace with str_starts_with() when PHP 8.0 is the minimum version
+			if (\strncmp($dirname, '.', 1) === 0) {
```

Comment:
```suggestion
			if (empty(\strncmp($dirname, '.', 1))) {
```

Reply:
`strncmp()` always returns an `int`, so we should check for this instead of a loose check.